main.tf

```
@@ -1,6 +1,7 @@
 locals {
-  kms_policy     = var.kms_key_arn != null ? data.aws_iam_policy_document.firehose_kms_role[0].json : null
-  parquet_prefix = "year=!{timestamp:yyyy}/month=!{timestamp:MM}/day=!{timestamp:dd}/hour=!{timestamp:HH}/"
+  kinesis = var.kinesis_arn != null ? { create = true } : {}
```

Having another look, why not something like this:

```suggestion
  kinesis = var.kinesis_arn != null ? [var.kinesis_arn] : []
```